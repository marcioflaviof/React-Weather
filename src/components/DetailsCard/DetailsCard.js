import { Box, Text } from "@theme-ui/components";
import { BsArrowUp, BsArrowDown } from "react-icons/bs";
import useCity from "../../hooks/useCity";

const weekText = {
  days: {
    marginLeft: "30px",
    fontWeight: "bold",
  },

  temp: {
    marginLeft: "25px",
    fontWeight: "bold",
    color: "orange",
  },
};

function DetailsCard({ sx }) {
  const [details] = useCity();

  return (
    <Box
      sx={{
        ...sx,
        background: "whitesmoke",
        maxWidth: ["300px", "450px"],
        height: "300px",
      }}
    >
      {details ? (
        <Box sx={{ padding: "10px 0px 0px 45px" }}>
          <Box
            sx={{
              padding: "20px 0px",
              fontSize: "2",
            }}
          >
            <Text variant="bold">
              {details.name}, {details.state} - {details.country}
            </Text>
          </Box>
          <Box
            sx={{
              fontSize: "6",
            }}
          >
            <Text variant="bold">
              {details.temp} {details.weather}
            </Text>
          </Box>
          <Box
            sx={{
              margin: "0px 0px 0px -5px",
              padding: "10px 0px",
            }}
          >
            <BsArrowDown color={"orange"} />
            <Text variant="bold">{details.min_temp}</Text>

            <BsArrowUp color={"orange"} />
            <Text variant="bold">{details.max_temp}</Text>
            <Text sx={{ padding: "0px 50px" }}>
              Sensação <strong>{details.sensation}</strong>
            </Text>
          </Box>
          <Box
            sx={{
              padding: "10px 0px",
            }}
          >
            <Text>
              Vento <strong>{details.wind} </strong>
            </Text>
            <Text sx={{ padding: "0px 20px" }}>
              Umidade <strong> {details.humidity} </strong>
            </Text>
          </Box>
          <Box
            sx={{
              margin: "10px 0px 15px -23px",
              background: "orange",
              opacity: "70%",
              height: "2px",
              width: "100%",
            }}
          />
          <Box>
            <Text variant="bold">{details.weekDays.dayOne.name}</Text>
            <Text sx={{ ...weekText.days }}>
              {details.weekDays.dayTwo.name}
            </Text>
            <Text sx={{ ...weekText.days }}>
              {details.weekDays.dayThree.name}
            </Text>
            <Text sx={{ ...weekText.days }}>
              {details.weekDays.dayFour.name}
            </Text>
            <Text sx={{ ...weekText.days }}>
              {details.weekDays.dayFive.name}
            </Text>
          </Box>
          <Box>
            <Text variant="bold" sx={{ color: "orange" }}>
              {details.weekDays.dayOne.min} {details.weekDays.dayOne.max}
            </Text>
            <Text sx={{ ...weekText.temp }}>
              {details.weekDays.dayTwo.min} {details.weekDays.dayTwo.max}
            </Text>
            <Text sx={{ ...weekText.temp }}>
              {details.weekDays.dayThree.min} {details.weekDays.dayThree.max}
            </Text>
            <Text sx={{ ...weekText.temp }}>
              {details.weekDays.dayFour.min} {details.weekDays.dayFour.max}
            </Text>
            <Text sx={{ ...weekText.temp }}>
              {details.weekDays.dayFive.min} {details.weekDays.dayFive.max}
            </Text>
          </Box>
        </Box>
      ) : (
        "Loading"
      )}
    </Box>
  );
}

export default DetailsCard;
