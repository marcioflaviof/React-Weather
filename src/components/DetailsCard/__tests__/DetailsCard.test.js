import React from "react";
import renderer from "react-test-renderer";
import DetailsCard from "../DetailsCard";

it("renders correctly", () => {
  const detailsCard = renderer.create(<DetailsCard />).toJSON();

  expect(detailsCard).toMatchSnapshot();
});
