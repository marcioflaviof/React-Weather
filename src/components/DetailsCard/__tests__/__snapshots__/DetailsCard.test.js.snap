// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-lg8ucf"
>
  <div
    className="css-ai50z5"
  >
    <span
      className="css-1p4nx4e"
    >
      Niterói
      , 
      RJ
       - 
      Brasil
    </span>
    <div
      className="css-6le2lh"
    >
      <span
        className="css-vurnku"
      >
        20ºC
         
        Nublado
      </span>
    </div>
    <div
      className="css-qqqbm1"
    >
      <svg
        color="orange"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "orange",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M4.646 9.646a.5.5 0 01.708 0L8 12.293l2.646-2.647a.5.5 0 01.708.708l-3 3a.5.5 0 01-.708 0l-3-3a.5.5 0 010-.708z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M8 2.5a.5.5 0 01.5.5v9a.5.5 0 01-1 0V3a.5.5 0 01.5-.5z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="css-vurnku"
      >
        16º
      </span>
      <svg
        color="orange"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "orange",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8 3.5a.5.5 0 01.5.5v9a.5.5 0 01-1 0V4a.5.5 0 01.5-.5z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M7.646 2.646a.5.5 0 01.708 0l3 3a.5.5 0 01-.708.708L8 3.707 5.354 6.354a.5.5 0 11-.708-.708l3-3z"
          fillRule="evenodd"
        />
      </svg>
      <span
        className="css-vurnku"
      >
        25º
      </span>
      <span
        className="css-116xqyd"
      >
        Sensação
      </span>
      <span
        className="css-vurnku"
      >
        19ºC
      </span>
    </div>
    <div
      className="css-so1jmj"
    >
      <span
        className="css-vurnku"
      >
        Vento
      </span>
      <span
        className="css-9ewue0"
      >
        18km/h
      </span>
      <span
        className="css-1l3oubh"
      >
        Umidade
      </span>
      <span
        className="css-vurnku"
      >
        89%
      </span>
    </div>
    <div
      className="css-1ujbi4u"
    />
    <div
      className="css-vurnku"
    >
      <span
        className="css-vurnku"
      >
        Terça
      </span>
      <span
        className="css-ltqn8"
      >
        Quarta
      </span>
      <span
        className="css-ltqn8"
      >
        Quinta
      </span>
      <span
        className="css-ltqn8"
      >
        Sexta
      </span>
      <span
        className="css-ltqn8"
      >
        Sábado
      </span>
    </div>
    <div
      className="css-vurnku"
    >
      <span
        className="css-1hjz83m"
      >
        18º
         
        26º
      </span>
      <span
        className="css-8d9naf"
      >
        18º
         
        28º
      </span>
      <span
        className="css-8d9naf"
      >
        19º
         
        30º
      </span>
      <span
        className="css-8d9naf"
      >
        23º
         
        35º
      </span>
      <span
        className="css-8d9naf"
      >
        23º
         
        37º
      </span>
    </div>
  </div>
</div>
`;
